eva/parser/lark_visitor/_functions.py

```diff
@@ -101,7 +102,11 @@ def create_udf(self, tree):
                 elif child.data == "udf_metadata":
                     # Each UDF metadata is a key value pair
                     key_value_pair = self.visit(child)
-                    metadata.append((key_value_pair[0].value, key_value_pair[1].value)),
+                    # value can be an integer or string
+                    value = key_value_pair[1]
+                    if isinstance(value, ConstantValueExpression):
```

Comment:
This should happen inside the visit imo (that's how it's done with the other nodes in this file). I think when you visit, if there is a constant value expression (decimal), then it should get the value and return it.

Reply:
I agree, but strings are returned as ConstantValueExpression objects, and integers are returned as int objects. I don't want to change the existing methods which is why this check is necessary.